tf-revoked-iteration/main.tf

```diff
@@ -12,11 +12,19 @@ data "hcp_packer_image" "ubuntu_us_east_2" {
 }
 
 resource "aws_instance" "app_server" {
-  count = data.hcp_packer_image.ubuntu_us_east_2.revoke_at == null ? 1 : 0
-
   ami           = data.hcp_packer_image.ubuntu_us_east_2.cloud_image_id
   instance_type = "t2.micro"
   tags = {
     Name = "Learn-HCP-Packer"
   }
+  
+  lifecycle {
+    precondition {
+      condition = try(
+        formatdate("YYYYMMDDhhmmss", data.hcp_packer_image.ubuntu_us_east_2.revoke_at) > formatdate("YYYYMMDDhhmmss", timestamp()),
+        data.hcp_packer_image.ubuntu_us_east_2.revoke_at == null
```

Comment:
To confirm my understanding. TF will try the expressions in order if one is true then no error is returned. Is that correct? 

If they both fail the try will fail and the error will be returned.

Reply:
Worth mentioning that `try` is specifically for this case, where one of the conditions may have an error, so it will return the result of the first one that is valid. If we change the order, `formatdate("YYYYMMDDhhmmss",...` will never be executed because `data.hcp_packer_image.ubuntu_us_east_2.revoke_at == null` never errors only and will always return `true` or `false`.